pimd/pim6_main.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * PIMv6 main()
+ * Copyright (C) 2021  David Lamparter for NetDEF, Inc.
+ * Copyright (C) 2008  Everton da Silva Marques (pim_main.c)
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+
+#include "lib/vrf.h"
+#include "lib/filter.h"
+#include "lib/plist.h"
+#include "lib/routemap.h"
+#include "lib/routing_nb.h"
+
+#include "lib/privs.h"
+#include "lib/sigevent.h"
+#include "lib/libfrr.h"
+#include "lib/version.h"
+
+#include "pimd.h"
+#include "pim_instance.h"
+#include "pim_errors.h"
+#include "pim_iface.h"
+#include "pim_zebra.h"
+
+zebra_capabilities_t _caps_p[] = {
```

Comment:
Not a good idea for 3 reasons:

- `pim6d` doesn't necessarily need the same capabilities. I think we can actually remove `ZCAP_BIND` since there is no MSDP in IPv6
- there is a pattern of having all these things together like in all other daemons, better to have it the same way
- `array_size()` doesn't work particularly well with `extern` array definitions in C (the array size would need to be copied into the header)